.github/workflows/ci.yml

```diff
@@ -64,20 +64,22 @@ jobs:
           python-version: ${{ matrix.python-version }}
           allow-prereleases: true
 
+      - name: Install uv
+        run: python -m pip install --upgrade uv
+
       - name: Install old NumPy
         if: matrix.numpy-version != 'latest'
-        run: python -m pip install numpy==${{ matrix.numpy-version }}
+        run: uv pip install --system numpy==${{ matrix.numpy-version }}
 
       - name: Install package
-        run: python -m pip install .[test]
+        run: uv pip install --system '.[test]'
 
       - name: Print NumPy version
         run: python -c 'import numpy as np; print(np.__version__)'
 
       - name: Test package
         run: >-
-          python -m pytest -ra --cov --cov-report=xml --cov-report=term
-          --durations=20
+          pytest -ra --cov --cov-report=xml --cov-report=term --durations=20
```

Comment:
As this is non-atomic, if it is preferred to not have it in this PR that's fine.